diff --git a/loopy/kernel/__init__.py b/loopy/kernel/__init__.py
index 68fcca1..d20dddf 100644
--- a/loopy/kernel/__init__.py
+++ b/loopy/kernel/__init__.py
@@ -1094,7 +1094,8 @@ class LoopKernel(ImmutableRecordWithoutPickling):
             "tags",
             "variables",
             "rules",
-            "statements",
+            "Statements",
+            "instructions",
             "Dependencies",
             "schedule",
             ])
@@ -1171,7 +1172,7 @@ class LoopKernel(ImmutableRecordWithoutPickling):
             for rule_name in natsorted(six.iterkeys(kernel.substitutions)):
                 lines.append(str(kernel.substitutions[rule_name]))
 
-        if "statements" in what:
+        if "Statements" in what or "instructions" in what:
             lines.extend(sep)
             if show_labels:
                 lines.append("STATEMENTS:")
